### ISSUE TITLE
Allow very confusing IAM setups (Should not require ListHostedZones permission?)

### ISSUE THREAD

**Issue by snarlysodboxer** [kubernetes/kops#743]
Tl:dr: Can't create cluster because AWS credential set doesn't have `ListHostedZones` rights, despite having rights to manage the required Route53 Hosted Zone.

Our credential set can assume a role from another AWS account which allows it to fully manage the Record Sets for their particular Hosted Zone for our cluster. This credential set is however, not allowed to ListHostedZones, since that would allow us to see the other account's other customers. (Transfer of domain is currently not an option.)

kops currently tries to `ListHostedZones` (I assume in order to test access to managing the Zone.) This causes the error `cannot find DNS Zone "mydomain.com".  Please pre-create the zone and set up NS records so that it resolves.` while trying to run `kops update cluster dev.k8s.mydomain.com`.

What if we passed the Zone ID instead of inferring it from the Zone Name, and used `ListResourceRecordSets` to confirm access?

P.S. You guys rock!

**Comment by snarlysodboxer** [comment]
Another thought is, a credential set can potentially have write access (`ChangeResourceRecordsSets`) to Records Sets without having read (`ListResourceRecordSets`) access. (We actually have a case where this is true.)

Maybe attempting a [write](http://docs.aws.amazon.com/Route53/latest/APIReference/API_ChangeResourceRecordSets.html) (`UPSERT`) to api.dev.k8s.mydomain.com would be sufficient? I'm sure others have more scope than I on the design decisions.

**Comment by chrislovecnm** [comment]
I think some of this was addressed with the changes in 1.5 IAM / Route53 IAM perms.  We have lots of discussion about IAM here https://github.com/kubernetes/kops/issues/1873.  Can we close this?
